src/python/pants/rules/core/filedeps.py

```diff
@@ -59,7 +59,7 @@ class Filedeps(Goal):
     ):
         unique_rel_paths.add(build_file_address.rel_path)
         adaptor = hydrated_target.adaptor
-        if hasattr(adaptor, "sources"):
```

Comment:
I'm assuming whoever implemented this didn't intentionally _not_ check that the sources resolved to something. i.e., I think it's fairly safe to assume that this stricter definition of `has_sources` will not make things not work that did previously.

Reply:
I agree :) I wrote a couple of the other instances of the "less-strict" check and that wasn't my intention, I only didn't realize it could be stricter.